src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -3860,23 +3853,35 @@ export interface CodingContract {
    *
    * Attempts to solve the Coding Contract with the provided solution.
    *
+   * There is an optional fourth parameter where you can assert the contract type.
+   * This will type the answer format you need to provide.
+   * Be aware that this function throws if the specified type does not match the type of the contract with that filename.
+   *
    * @example
    * ```js
-   * const reward = ns.codingcontract.attempt(yourSolution, filename, hostname);
+   * const reward = ns.codingcontract.attempt("solution as a string", filename, hostname);
+   * // or
+   * const reward = ns.codingcontract.attempt(["answer", "in", "the", "correct", "format"], filename, hostname);
```

Comment:
```suggestion
   * const reward = ns.codingcontract.attempt(["answer", "as", "an", "array"], filename, hostname);
```
"Correct format" is a bit too heavy of a value judgement imo